ui/raidboss/emulator/data/LogEventHandler.ts

```diff
@@ -0,0 +1,130 @@
+import EmulatorCommon from '../EmulatorCommon';
+import EventBus from '../EventBus';
+import { Lang } from 'types/global';
+import LineEvent from './network_log_converter/LineEvent';
+import { LineEvent0x01 } from './network_log_converter/LineEvent0x01';
+
+export default class LogEventHandler extends EventBus {
+  static doesLineMatch(line: string,
+      regexes: Record<Lang, RegExp> | RegExp): RegExpExecArray | null {
+    if (regexes instanceof RegExp)
+      return regexes.exec(line);
+
+    for (const langStr in regexes) {
+      const lang = langStr as keyof typeof regexes;
+      const res = regexes[lang].exec(line);
+      if (res) {
+        if (res.groups)
+          res.groups.language = lang;
+        return res;
+      }
+    }
+    return null;
+  }
+
+  static isMatchStart(line: string): RegExpMatchArray | false {
```

Comment:
It's used in `Encounter`. It lives in `LogEventHandler` because `isMatchEnd` also lives there and is used there. Given that all of these are using regexes from `EmulatorCommon`, maybe it makes sense to move them both there?

Reply:
That seems reasonable!

Reply:
This change is done.